cmd/blockchaincmd/convert.go

```diff
@@ -418,13 +422,44 @@ func convertSubnetToL1(
 	controlKeysList,
 	subnetAuthKeysList []string,
 	validatorManagerAddressStr string,
-) ([]*txs.ConvertSubnetToL1Validator, bool, error) {
+	doStrongInputsCheck bool,
+) ([]*txs.ConvertSubnetToL1Validator, bool, bool, error) {
+	if subnetID == ids.Empty {
+		return nil, false, false, clierrors.ErrNoSubnetID
+	}
+	if blockchainID == ids.Empty {
+		return nil, false, false, clierrors.ErrNoBlockchainID
+	}
+	if !common.IsHexAddress(validatorManagerAddressStr) {
+		return nil, false, false, clierrors.ErrInvalidValidatorManagerAddress
+	}
 	avaGoBootstrapValidators, err := ConvertToAvalancheGoSubnetValidator(bootstrapValidators)
 	if err != nil {
-		return avaGoBootstrapValidators, false, err
+		return avaGoBootstrapValidators, false, false, err
 	}
 	deployer.CleanCacheWallet()
 	managerAddress := common.HexToAddress(validatorManagerAddressStr)
+
+	if doStrongInputsCheck {
+		ux.Logger.PrintToUser("You are about to create a ConvertSubnetToL1Tx on %s with the following content:", network.Name())
+		ux.Logger.PrintToUser("  Subnet ID: %s", subnetID)
+		ux.Logger.PrintToUser("  Blockchain ID: %s", blockchainID)
+		ux.Logger.PrintToUser("  Manager Address: %s", managerAddress.Hex())
+		ux.Logger.PrintToUser("  Validators:")
+		for _, val := range bootstrapValidators {
+			ux.Logger.PrintToUser("    %s", val.NodeID)
+			ux.Logger.PrintToUser("    %.5f", float64(val.Balance)/float64(units.Avax))
+		}
+		ux.Logger.PrintToUser("")
+		ux.Logger.PrintToUser("Please review the details of the ConvertSubnetToL1 Transaction")
+		ux.Logger.PrintToUser("")
+		if doContinue, err := app.Prompt.CaptureYesNo("Do you want to create the transaction?"); err != nil {
+			return avaGoBootstrapValidators, false, false, err
+		} else if !doContinue {
+			return avaGoBootstrapValidators, true, false, err
+		}
```

Comment:
doContinue, err := app.Prompt.CaptureYesNo("Do you want to create the transaction?")
return avaGoBootstrapValidators, doContinue, false, err

Reply:
I believe it is cleaner this way